### ISSUE TITLE
`PartitionedTableProxy` is missing table methods, leading to a poor UX

### ISSUE THREAD

**Issue by chipkent** [deephaven/deephaven-core#2890]
`PartitionedTableProxy` is missing table methods, which results in a very poor user experience.

Consider this case.  `merge` and `drop_columns` are missing from `PartitionedTableProxy`.  As a result, I had to:
1. Explicitly add `.target` 
2. Move `merge` below `.target`
3. Move `drop_columns` below `.target`

```python
from deephaven import time_table
from deephaven.updateby import ema_time_decay

t1 = time_table("00:00:01").update(["Label = ii%2", "X = ii%2 == 0 ? cos(0.1*ii) : sin(0.01*ii)"])

t2 = t1.partition_by(["Label"]).proxy() \
    .update(["Data=X_.subVector(ii-14,ii)"]) \
    .update(["H=max(Data)"]) \
    .update(["L=min(Data)"]) \
    .update(["ATR=max(H-L, abs(H-X), abs(L-X))"]) \
    .target \
    .merge() \
    .drop_columns("Data")
```

These rough edges break the user view that partitioned tables are "just tables broken into smaller parts that can be operated on separately and then recombined".

**Comment by rcaudy** [comment]
- Proxies behave as `TableOperations`. It's a more limited interface than `Table`, because you *cannot* interrogate for schema, or try to pull data out. If you'd like to add more methods to `TableOperations` (like `dropColumns`), write an issue for that and we can debate/prioritize. Note that this is the same interface used for client-side proxies. We intentionally constrain the operations to those that make sense when proxied.
- `merge` isn't a `Table` method, it's a `PartitionedTable` method. I strongly dislike the idea of muddying the interface layers by making `merge()` work on a proxy directly. I spent years cleaning up after similar "convenience" functions in other places.

**Comment by chipkent** [comment]
Here is an organized list of what is not currently supported:

**Should support**
- where_one_of
- lazy_update
- tail_by
- head_by
- head_pct
- tail_pct
- ungroup
- drop_columns
- rename_columns
- move_columns
- move_columns_up
- move_columns_down

**Should not support:**
- size
- columns
- coalesce
- restrict_sort_to
- to_string
- meta_table

**Probably not support**
- partition_by
- snapshot_history

**Not sure if we should support (probably not)**
- format_columns
- format_column_where
- format_row_where
- layout_hints

**Comment by rcaudy** [comment]
I think we can mostly agree on this categorization. A few notes:
- `whereOneOf` is not a Java Table method anymore. I wouldn't add it to `TableOperations`.
- `restrictsortTo` might be OK to add, although I would like to jettison it in favor of better tools.
- `partitionBy` and `snapshotHistory` are probably fine to add.
- `format` methods could probably be added, although I'm not sure how important they are.

I'm not intending to bubble this up to the top of the to-do list very soon.

**Comment by chipkent** [comment]
Updated based on @rcaudy comments:
Here is an organized list of what is not currently supported:

**Should support**
- lazy_update
- tail_by
- head_by
- head_pct
- tail_pct
- ungroup
- drop_columns
- rename_columns
- move_columns
- move_columns_up
- move_columns_down
- partition_by
- snapshot_history

**Should support (lower priority)**
- format_columns
- format_column_where
- format_row_where
- layout_hints

**Should not support:**
- size
- columns
- coalesce
- restrict_sort_to
- to_string
- meta_table
- where_one_of

**Comment by rbasralian** [comment]
To me, `ungroup()`, `lazy_update()`, `drop_columns()`, and `rename_columns()` are the most important — all four come up all the time. The first two you can't really do any other way, and second two are extremely common and difficult to live without. (`move_columns()` you can harmlessly do after merging later on, and head/tail_by/pct I just don't use as often.)

**Comment by jjbrosnan** [comment]
I agree with Chip's list, though I think that the `move_columns`, `move_columns_up`, and `move_columns_down` should be shifted to the lower priority list. I don't think I've ever really needed to use those. I'm sure there are cases where they're useful, but they're (IMO) definitely not as important as the first several in the list.

**Comment by devinrsmith** [comment]
I can start working on plumbing the java TableOperations side of things. At least some already exist at that layer, and just need the final wrapping in python (lazy_update for example).

**Comment by devinrsmith** [comment]
There's probably a larger discussion we should have around `move_columns`, `rename_columns`, `drop_columns` and variants - at least from the gRPC layer, I doubt we want to have a bunch of new messages types - we could probably do with generalizing it into a single method where you can accomplish multiple things. And then way also have to ask how it compares to `view` / `update_view`.
